docs/reference/aggregations/bucket/frequent-items-aggregation.asciidoc

```diff
@@ -0,0 +1,159 @@
+[[search-aggregations-bucket-frequent-items-aggregation]]
+=== Frequent items aggregation
+++++
+<titleabbrev>Frequent items</titleabbrev>
+++++
+
+A bucket aggregation which finds frequent item sets. Finding frequent item sets 
+is a form of association rules mining that identifies items that are often occur 
+together. Items that are frequently purchased together or log events that tend 
+to co-occur are examples of frequent item sets. Finding frequent item sets helps 
+to discover relationships between different data points (items) in a data set 
+and to create association rules based on those relationships.
+
```

Comment:
We should mention that `frequent_items` reports "closed itemsets":

"A frequent item set is called closed if no superset has the same support."

That's very technical, you maybe find a better way to formulate it. It simply means that if you have candidates like:

cat, dog, crocodile -> 0.2
cat, dog, crocodile, river -> 0.2

only cat, dog, crocodile, river gets returned, the subset cat, dog, crocodile gets skipped. On the other side for the sets: 

cat, doc -> 0.4
cat, dog, crocodile -> 0.2
cat, dog, crocodile, river -> 0.2

Both cat, doc -> 0.4 and cat, dog, crocodile, river -> 0.2 are returned, because cat, doc has a higher/different support.